lib/utils.pm

```diff
@@ -1825,4 +1827,28 @@ sub install_patterns {
     }
 }
 
+sub common_service_action {
+    my ($service, $type, $action) = @_;
+
+    if ($type eq 'SystemV') {
+        if ($action eq 'on') {
+            assert_script_run 'chkconfig ' . $service . ' on';
+            assert_script_run 'chkconfig ' . $service;
+        } else {
+            assert_script_run '/etc/init.d/' . $service . ' ' . $action;
+            assert_script_run 'service ' . $service . ' status';
+        }
+    }
+    elsif ($type eq 'Systemd') {
+        systemctl $action . ' ' . $service;
+        systemctl 'status ' . $service;
+    } else {
+        die "Unsupported service type, please check it again.";
+    }
+}
+
+sub get_service_type {
```

Comment:
you have service_type within service_check.pm, but here you create another function?

Reply:
changed to use service_check.pm -> module's full function check.